flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/generated/BoundComparator.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.generated;
+
+import org.apache.flink.table.dataformat.BaseRow;
+
+import java.io.Serializable;
+
+/**
+ * comparing boundary values.
+ */
+public interface BoundComparator extends Serializable {
```

Comment:
I think more comments are needed for this interface, I really cannot get the point what's this interface is used for, and what the interface means...

Reply:
For example, 
`SELECT d, e, sum(e) over (partition by d order by e desc range between 5 PRECEDING and 4 PRECEDING) FROM Table`
The above SQL is ranged according to the `e` field.
The previous 5 PRECEDING means that the `e` value of current Row should be greater than or equal to the `e` value of range header plus 5.
So the `BoundComparator` is used to code generate this logic.